test/e2e/autoscaler/autoscaler_test.go

```diff
@@ -0,0 +1,990 @@
+						highPriorityRunning++
+					} else if strings.Contains(pod.Name, lowPriorityApp) {
+						lowPriorityRunning++
+					}
+				}
+			}
+			// In resource-constrained environments, we may not see perfect preemption
+			return highPriorityRunning == 4 && lowPriorityRunning <= 3
+		}, 60*time.Second, 3*time.Second).Should(gomega.BeTrue(), "High-priority pods should scale to 4, low-priority should be preempted or stay same")
+	})
+
+	// Test resource competition between multiple autoscaling deployments
+	It("Verify_Multi_Queue_Autoscaling_With_Resource_Competition", func() {
+		By("Testing resource competition between multiple autoscaling deployments in different queues")
+
+		app1Name := "queue1-app-" + common.RandSeq(5)
+		app2Name := "queue2-app-" + common.RandSeq(5)
+		app3Name := "queue3-app-" + common.RandSeq(5)
+
+		// Create deployments in different queues with different priorities
+		deployment1 := createTestDeployment(app1Name, ns, 2, map[string]string{
+			"cpu":    "100m",
+			"memory": "128Mi",
+		})
+		deployment1.Spec.Template.Annotations = map[string]string{
+			"yunikorn.apache.org/queue": "root.production",
+		}
+
+		deployment2 := createTestDeployment(app2Name, ns, 2, map[string]string{
+			"cpu":    "150m",
+			"memory": "192Mi",
+		})
+		deployment2.Spec.Template.Annotations = map[string]string{
+			"yunikorn.apache.org/queue": "root.staging",
+		}
+
+		deployment3 := createTestDeployment(app3Name, ns, 2, map[string]string{
+			"cpu":    "50m",
+			"memory": "64Mi",
+		})
+		deployment3.Spec.Template.Annotations = map[string]string{
+			"yunikorn.apache.org/queue": "root.development",
+		}
+
+		// Deploy all applications
+		deploymentObj1, err := kClient.CreateDeployment(deployment1, ns)
+		Ω(err).NotTo(HaveOccurred())
+		defer func() { _ = kClient.DeleteDeployment(deploymentObj1.Name, ns) }() //nolint:errcheck // Cleanup operation
+
+		deploymentObj2, err := kClient.CreateDeployment(deployment2, ns)
+		Ω(err).NotTo(HaveOccurred())
+		defer func() { _ = kClient.DeleteDeployment(deploymentObj2.Name, ns) }() //nolint:errcheck // Cleanup operation
+
+		deploymentObj3, err := kClient.CreateDeployment(deployment3, ns)
+		Ω(err).NotTo(HaveOccurred())
+		defer func() { _ = kClient.DeleteDeployment(deploymentObj3.Name, ns) }() //nolint:errcheck // Cleanup operation
+
+		By("Verifying initial deployment state")
+		gomega.Eventually(func() bool {
+			podsInNs, getErr := kClient.GetPods(ns)
+			if getErr != nil {
+				return false
+			}
+			app1Running := 0
+			app2Running := 0
+			app3Running := 0
+
+			for _, pod := range podsInNs.Items {
+				if pod.Status.Phase == v1.PodRunning {
+					switch pod.Labels["app"] {
+					case app1Name:
+						app1Running++
+					case app2Name:
+						app2Running++
+					case app3Name:
+						app3Running++
+					}
+				}
+			}
+
+			return app1Running >= 2 && app2Running >= 2 && app3Running >= 2
+		}, 180*time.Second, 5*time.Second).Should(gomega.BeTrue())
```

Comment:
Again, same behavior can be achieved with the new helper methods.
3 apps, 3 method calls:
kClient.WaitForPodCountBySelectorRunning(ns, "app=app1", 2, 60)
kClient.WaitForPodCountBySelectorRunning(ns, "app=app2", 2, 60)
kClient.WaitForPodCountBySelectorRunning(ns, "app=app3", 2, 60)